src/TwigComponent/src/DependencyInjection/Compiler/TwigComponentPass.php

```diff
@@ -27,23 +28,31 @@ public function process(ContainerBuilder $container): void
     {
         $serviceIdMap = [];
 
-        foreach (array_keys($container->findTaggedServiceIds('twig.component')) as $serviceId) {
-            $definition = $container->getDefinition($serviceId);
+        foreach ($container->getDefinitions() as $id => $definition) {
+            $class = $definition->getClass();
+
+            if (!class_exists($class, false) || !$attribute = TwigComponent::forClass($class)) {
+                continue;
+            }
+
+            $name = $attribute->getName();
```

Comment:
In case you want to support doctrine annotations as well, you could add the following (if no name was defined in the tag)

```php
private function getNameFromDoctrineAnnotation(ContainerBuilder $container, Definition $definition): ?string
    {
        if (!$container->has(Reader::class)) {
            return null;
        }

        /** @var Reader $reader */
        $reader = $container->get(Reader::class);
        if ($reader === null) {
            return null;
        }

        $reflectionClass = new ReflectionClass($definition->getClass());

        /** @var Component|null $componentAnnotation */
        $componentAnnotation = $reader->getClassAnnotation($reflectionClass, Component::class);

        if ($componentAnnotation === null) {
            return null;
        }

        return $componentAnnotation->name;
    }
```